js/binance.js

```diff
@@ -4894,4 +4894,47 @@ module.exports = class binance extends Exchange {
     async addMargin (symbol, amount, params = {}) {
         return await this.modifyMarginHelper (symbol, amount, 1, params);
     }
+
+    async getMarginInterestRate (currency, since = undefined, limit = undefined, tier = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (limit > 100) {
+            throw new BadRequest (this.id + ' getMarginInterestRate limit parameter cannot exceed 100');
+        } else if (!limit) {
+            limit = 20;
+        }
+        const request = {
+            'asset': this.safeCurrencyCode (currency),
+            'limit': limit,
+        };
+        if (tier) {
+            request['vipLevel'] = tier;
+        }
+        if (since) {
+            const sinceDate = new Date (since);
+            const threeMonthsAgo = new Date ();
+            threeMonthsAgo.setMonth (threeMonthsAgo.getMonth () - 3);
+            threeMonthsAgo.setDate (threeMonthsAgo.getDate ());
+            threeMonthsAgo.setHours (0, 0, 0);
+            if (sinceDate <= threeMonthsAgo) {
+                throw new BadRequest (this.id + ' getMarginInterestRate since parameter cannot pre-date today - 3 months');
+            }
+            request['startTime'] = since;
+            const endTime = this.sum (since, limit * 86400000);
+            request['endTime'] = Math.min (endTime, this.milliseconds ());
```

Comment:
I tried replacing lines 4922 and 4923 with

```
const endTime = Precise.stringAdd (since.toString (), Precise.stringMul (limit.toString () * '86400000'));
request['endTime'] = parseInt (Precise.stringMin (endTime, this.milliseconds ().toString ()));
```

but this gives me incorrect results from the server, where the response only has one item